### PR TITLE
feat: eventemitter types

### PR DESCRIPTION
**Pull request by pavanjoshi914**

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -271,6 +271,9 @@ interface WebLNProvider {
   sendPayment(paymentRequest: string): Promise<SendPaymentResponse>;
   signMessage(message: string): Promise<SignMessageResponse>;
   verifyMessage(signature: string, message: string): Promise<void>;
+  on(eventName: string, listener: () => void): this;
+  off(eventName: string, listener: () => void): this;
+  emit(eventName: string, ...args: any[]): boolean;
```
Comment (rolznz) [inline types/index.d.ts]: emit should not be called externally, can you remove this one?

Comment (pavanjoshi914) [reply types/index.d.ts]: resolved

Comment (rolznz) [reply types/index.d.ts]: @pavanjoshi914 thanks.

Sorry, I just realized, on and off should also be optional.

Comment (pavanjoshi914) [reply types/index.d.ts]: nice catch! thanks. made those changes

**Comment by bumi** [comment]
this requires a npm package release, right?

**Comment by rolznz** [comment]
@bumi I deployed one immediately after I merged it.

@pavanjoshi914 looking again I think we want to return `Promise<void>` instead of `this`. We cannot be consistent with the events API because our one requires awaiting enable().

**Comment by bumi** [comment]
@rolznz ok, thanks. (then the package.json is old in master)

**Pull request merged**